src/v2/Components/ArtworkFilter/ArtworkFilters/__tests__/SizeFilter.jest.tsx

```diff
@@ -185,3 +227,116 @@ describe("SizeFilter", () => {
+
+  it("should return predefined sizes in inches when `in` metric is specified", () => {
+    expect(getPredefinedSizesByMetric("in")).toEqual(SIZES_IN_INCHES)
+  })
+})
+
+describe("getUnitLabelByMetric", () => {
+  it("should return centimeters label when `cm` metric is specified", () => {
+    expect(getUnitLabelByMetric("cm")).toBe("cm")
+  })
+
+  it("should return inches label when `in` metric is specified", () => {
+    expect(getUnitLabelByMetric("in")).toBe("in")
+  })
+})
+
+describe("parseRange", () => {
+  it("correctly parse range in inches", () => {
+    expect(parseRange("10-20", "in")).toEqual([10, 20])
+  })
+
+  it("correctly parse range in inches with default value", () => {
+    expect(parseRange("10-*", "in")).toEqual([10, "*"])
+  })
+
+  it("correctly parse range in inches when all values are default", () => {
+    expect(parseRange("*-*", "in")).toEqual(["*", "*"])
+  })
+
+  it("correctly parse range in centimeters", () => {
+    expect(parseRange("1-2", "cm")).toEqual([3, 5])
+  })
+
+  it("correctly parse range in centimeters with default value", () => {
+    expect(parseRange("1-*", "cm")).toEqual([3, "*"])
+  })
+
+  it("correctly parse range in centimeters when all values are default", () => {
+    expect(parseRange("*-*", "cm")).toEqual(["*", "*"])
+  })
+})
+
+describe("getCustomSizeRangesInInches", () => {
+  it("should return correct custom size ranges in inches", () => {
+    const range: CustomSize = {
+      width: [5, 10],
+      height: [15, 20],
+    }
+    const result = {
+      width: "5-10",
+      height: "15-20",
+    }
+
+    expect(getCustomSizeRangesInInches(range, "in")).toEqual(result)
+  })
+
+  it("should return correct custom size ranges in inches when default values are passed", () => {
+    const range: CustomSize = {
+      width: [5, "*"],
+      height: ["*", 20],
+    }
+    const result = {
+      width: "5-*",
+      height: "*-20",
+    }
+
+    expect(getCustomSizeRangesInInches(range, "in")).toEqual(result)
+  })
+
+  it("should return custom size ranges with default values", () => {
+    const range: CustomSize = {
+      width: ["*", "*"],
+      height: ["*", "*"],
+    }
+    const result = {
+      width: "*-*",
+      height: "*-*",
+    }
+
+    expect(getCustomSizeRangesInInches(range, "in")).toEqual(result)
+  })
+
+  it("should return correct custom size ranges in centimeters", () => {
+    const range: CustomSize = {
+      width: [5, 10],
+      height: [15, 20],
+    }
+    const result = {
+      width: "1.968503937007874-3.937007874015748",
+      height: "5.905511811023622-7.874015748031496",
+    }
+
+    expect(getCustomSizeRangesInInches(range, "cm")).toEqual(result)
+  })
+
+  it("should return correct custom size ranges in centimeters when default values are passed", () => {
+    const range: CustomSize = {
+      width: ["*", 10],
+      height: [15, "*"],
+    }
+    const result = {
+      width: "*-3.937007874015748",
+      height: "5.905511811023622-*",
```

Comment:
**suggestion**: Related to the above comment - I think we can trim off some precision here.